test/bdd/pkg/vc/vc_steps.go

```diff
@@ -163,8 +181,18 @@ func (e *Steps) storeCredential(storeVCRequestArgKey string) error {
 	return nil
 }
 
-func (e *Steps) retrieveCredential(credentialID, profileName, expectedIssuerID, expectedIssuerName string) error {
-	escapedCredentialID := url.PathEscape(credentialID)
+func (e *Steps) retrieveCredential(profileName string) error {
+	vcMap, err := getVCMap(e.bddContext.CreatedCredential)
+	if err != nil {
+		return err
+	}
+
+	vcID, found := vcMap["id"].(string)
```

Comment:
this also covers the case when id is present but of wrong type so error message is bit misleading

Reply:
You're right. I separated the map get and the type assertion into separate lines so I can check both errors separately. I also refactored the other type assertion I do in that file so that I can check the two errors separately.